ros/src/waypoint_updater/waypoint_updater.py

```diff
@@ -74,11 +75,40 @@ def get_closest_waypoint_idx(self):
         return closest_idx
 
     def publish_waypoints(self, closest_idx):
+        final_lane = self.generate_lane()
+        self.final_waypoints_pub.publish(final_lane)
+
+        # lane = Lane()
+        # lane.header = self.base_waypoints.header
+        # lane.waypoints = self.base_waypoints.waypoints[closest_idx:closest_idx + LOOKAHEAD_WPS]
+        # rospy.loginfo("closest_idx %s " % closest_idx)
+        # self.final_waypoints_pub.publish(lane)
+
+    def generate_lane(self):
         lane = Lane()
-        lane.header = self.base_waypoints.header
-        lane.waypoints = self.base_waypoints.waypoints[closest_idx:closest_idx + LOOKAHEAD_WPS]
-        rospy.loginfo("closest_idx %s " % closest_idx)
-        self.final_waypoints_pub.publish(lane)
+        closest_idx = self.get_closest_waypoint_idx()
+        farthest_idx = closest_idx + LOOKAHEAD_WPS
+        base_waypoints = self.base_lane.waypoints[closest_idx:farthest_idx]
```

Comment:
`self.base_lane` is not specified in `__init__()`

Reply:
nice catch! thanks